app/channels/link_channel.rb

```diff
@@ -3,37 +3,35 @@ def subscribed
     if params[:id].present?
       link = Link.find(params[:id])
       if link
-        if connection&.watched_link
-          connection.watched_link.live_client_started_at = nil
-          connection.watched_link.save
-        end
-        connection.watched_link = link
         link.live_client_started_at = Time.now
         link.save
+
+        connection&.watched_links.push(link)
         stream_from "Link::#{params[:id]}"
       end
     end
   end
 
   def unsubscribed
-    if connection&.watched_link
-      connection.watched_link.live_client_started_at = nil
-      connection.watched_link.save
+    if params[:id].present?
+      link = Link.find(params[:id])
+      if link
+        stop_stream_from "Link::#{params[:id]}"
```

Comment:
So, in my testing, this works totally fine, but stop_stream_for might be a bit more rails-y

https://api.rubyonrails.org/v7.1.1/classes/ActionCable/Channel/Streams.html#method-i-stop_stream_for

Reply:
Ah, like `stop_stream_for link`?